todo/src/components/DisplayTodos.js

```diff
@@ -0,0 +1,64 @@
+import React from "react";
+import { useSelector, useDispatch } from "react-redux";
+import { completeTodo } from "../actions/todoActions";
+import VisibilityButtons from "./VisibilityButtons";
+
+const DisplayTodos = () => {
+  const todoList = useSelector((state) => Object.values(state.toDos));
+  const visibility = useSelector((state) => state.visibilityFilter);
+
+  const dispatch = useDispatch();
+
+  const handleClick = (e) => {
+    e.preventDefault();
+    let id = e.currentTarget.id;
+    dispatch(completeTodo(id));
+  };
+
+  const filterVisibility = () => {
```

Comment:
You can use the visibility variable to `filter` your todos based on its value. This way you don't have to loop through your todos three times. Just `uno`